test/AutoRest.TestServer.Tests/Infrastructure/TestServerLowLevelTestBase.cs

```diff
@@ -81,8 +81,13 @@ private async Task Test(string scenario, Func<Uri, Task> test, bool ignoreScenar
         private static string GetScenarioName()
         {
             var testName = TestContext.CurrentContext.Test.Name;
-            var indexOfUnderscore = testName.IndexOfAny(new[]{'_', '('});
-            return indexOfUnderscore == -1 ? testName : testName[..indexOfUnderscore];
+            var indexOfDelimiter = testName.LastIndexOf('_');
+            if (indexOfDelimiter == -1)
+            {
+                indexOfDelimiter = testName.IndexOf('(');
```

Comment:
Previously, test names do not contain `_`.  Now the testserver has `DPGAddOptionalInput_NoParams`, `DPGAddOptionalInput_RequiredOptionalParam`, `DPGAddOptionalInput_OptionalParam` as test names. Instead of passing `DPGAddOptionalInput` as the name to report coverage, now we need to pass the full name. 

As a workaround to not break previous tests, I made this change and we will name the test like `DpgAddOptionalInput_NoParams_Test` and then `DpgAddOptionalInput_NoParams` will be passed as the name for coverage to match the full test name.